hpedockerplugin/request_context.py

```diff
@@ -0,0 +1,567 @@
+        if 'Opts' in contents and contents['Opts']:
+            received_opts = contents.get('Opts').keys()
+            diff = set(mutually_exclusive_ops) - set(received_opts)
+            if len(diff) < len(mutually_exclusive_ops) - 1:
+                mutually_exclusive_ops.sort()
+                msg = "Operations %s are mutually exclusive and cannot be " \
+                      "specified together. Please check help for usage." % \
+                      mutually_exclusive_ops
+                raise exception.InvalidInput(reason=msg)
+
+    @staticmethod
+    def _validate_opts(operation, contents, valid_opts, mandatory_opts=None):
+        LOG.info("Validating options for operation '%s'" % operation)
+        if 'Opts' in contents and contents['Opts']:
+            received_opts = contents.get('Opts').keys()
+
+            if mandatory_opts:
+                diff = set(mandatory_opts) - set(received_opts)
+                if diff:
+                    # Print options in sorted manner
+                    mandatory_opts.sort()
+                    msg = "One or more mandatory options %s are missing " \
+                          "for operation %s" % (mandatory_opts, operation)
+                    LOG.error(msg)
+                    raise exception.InvalidInput(reason=msg)
+
+            diff = set(received_opts) - set(valid_opts)
+            if diff:
+                diff = list(diff)
+                diff.sort()
+                msg = "Invalid option(s) %s specified for operation %s. " \
+                      "Please check help for usage." % \
+                      (diff, operation)
+                LOG.error(msg)
+                raise exception.InvalidInput(reason=msg)
+
+
+class FileRequestContextBuilder(RequestContextBuilder):
+    def __init__(self, backend_configs):
+        super(FileRequestContextBuilder, self).__init__(backend_configs)
+
+    def _get_build_req_ctxt_map(self):
+        build_req_ctxt_map = OrderedDict()
+        # If share-dir is specified, file-store MUST be specified
+        build_req_ctxt_map['filePersona'] = \
+            self._create_share_req_ctxt
+        # build_req_ctxt_map['persona,cpg'] = \
+        #     self._create_share_req_ctxt
+        # build_req_ctxt_map['persona,cpg,size'] = \
+        #     self._create_share_req_ctxt
+        # build_req_ctxt_map['persona,cpg,size,fpg_name'] = \
+        #     self._create_share_req_ctxt
+        build_req_ctxt_map['virtualCopyOf,shareName'] = \
+            self._create_snap_req_ctxt
+        build_req_ctxt_map['updateShare'] = \
+            self._create_update_req_ctxt
+        build_req_ctxt_map['help'] = self._create_help_req_ctxt
+        return build_req_ctxt_map
+
+    def _create_share_req_params(self, name, options):
+        LOG.info("_create_share_req_params: Entering...")
+        # import pdb
+        # pdb.set_trace()
+        backend = self._get_str_option(options, 'backend', 'DEFAULT')
+        config = self._backend_configs[backend]
```

Comment:
This might result in error for following use case.
Consider A and B are block storage backends.
C and D are File Storage backends.

Now create share with -o backend=A

Reply:
User will get the following message:
`
"ERROR:` Backend 'A' was NOT initialized successfully. Please check hpe.conf for incorrect entries and rectify it."
`
We will need to enhance the message as:
"ERROR:` Backend 'A' was either NOT initialized successfully or not configured. Please check hpe.conf for incorrect entries and rectify it."
`
However, I see another issue with this - after merging both block and file configurations, we had to introduce "DEFAULT_BLOCK" and "DEFAULT_FILE". While initializing backend in the context, we need to ensure that appropriate DEFAULT _XXX is assigned based on block or file. Will take are of this.